plugins/kinetic-dp/fu-kinetic-dp-device.c

```diff
@@ -0,0 +1,213 @@
+fu_kinetic_dp_device_chip_id_to_family(FuKineticDpChip chip_id)
+{
+	if (chip_id == FU_KINETIC_DP_CHIP_PUMA_2900 || chip_id == FU_KINETIC_DP_CHIP_PUMA_2920)
+		return FU_KINETIC_DP_FAMILY_PUMA;
+	if (chip_id == FU_KINETIC_DP_CHIP_MUSTANG_5200)
+		return FU_KINETIC_DP_FAMILY_MUSTANG;
+	if (chip_id == FU_KINETIC_DP_CHIP_JAGUAR_5000)
+		return FU_KINETIC_DP_FAMILY_JAGUAR;
+	return FU_KINETIC_DP_FAMILY_UNKNOWN;
+}
+
+static const gchar *
+fu_kinetic_dp_device_get_name_for_chip_id(FuKineticDpChip chip_id)
+{
+	if (chip_id == FU_KINETIC_DP_CHIP_JAGUAR_5000)
+		return "KTM50X0";
+	if (chip_id == FU_KINETIC_DP_CHIP_MUSTANG_5200)
+		return "KTM52X0";
+	if (chip_id == FU_KINETIC_DP_CHIP_PUMA_2900)
+		return "MC2900";
+	return NULL;
+}
+
+gboolean
+fu_kinetic_dp_device_dpcd_read_oui(FuKineticDpDevice *self,
+				   guint8 *buf,
+				   gsize bufsz,
+				   GError **error)
+{
+	if (bufsz < DPCD_SIZE_IEEE_OUI) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_INVALID_DATA,
+			    "aux dpcd read buffer size [0x%x] is too small to read IEEE OUI",
+			    (guint)bufsz);
+		return FALSE;
+	}
+	if (!fu_dpaux_device_read(FU_DPAUX_DEVICE(self),
+				  DPCD_ADDR_IEEE_OUI,
+				  buf,
+				  DPCD_SIZE_IEEE_OUI,
+				  FU_KINETIC_DP_DEVICE_TIMEOUT,
+				  error)) {
+		g_prefix_error(error, "aux dpcd read OUI failed: ");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+gboolean
+fu_kinetic_dp_device_dpcd_write_oui(FuKineticDpDevice *self, const guint8 *buf, GError **error)
+{
+	if (!fu_dpaux_device_write(FU_DPAUX_DEVICE(self),
+				   DPCD_ADDR_IEEE_OUI,
+				   buf,
+				   DPCD_SIZE_IEEE_OUI,
+				   FU_KINETIC_DP_DEVICE_TIMEOUT,
+				   error)) {
+		g_prefix_error(error, "aux dpcd write OUI failed: ");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+fu_kinetic_dp_device_setup(FuDevice *device, GError **error)
+{
+	FuKineticDpDevice *self = FU_KINETIC_DP_DEVICE(device);
+	FuKineticDpDevicePrivate *priv = GET_PRIVATE(self);
+	const gchar *chip_id_str;
+
+	/* FuDpauxDevice->setup */
+	if (!FU_DEVICE_CLASS(fu_kinetic_dp_device_parent_class)->setup(device, error))
+		return FALSE;
+
+	/* sanity check */
+	if (fu_dpaux_device_get_dpcd_ieee_oui(FU_DPAUX_DEVICE(device)) == 0x0) {
+		g_set_error_literal(error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED, "no IEEE OUI set");
+		return FALSE;
+	}
+
+	/* set up the device name */
+	chip_id_str = fu_kinetic_dp_device_get_name_for_chip_id(priv->chip_id);
+	if (chip_id_str != NULL)
+		fu_device_set_name(FU_DEVICE(self), chip_id_str);
+
+	/* detect chip family */
+	priv->family = fu_kinetic_dp_device_chip_id_to_family(priv->chip_id);
+	/* TODO set max firmware size base upon family if we need it */
+	switch (priv->family) {
+	case FU_KINETIC_DP_FAMILY_JAGUAR:
+		/* TODO: set max firmware size for Jaguar */
+		break;
+	case FU_KINETIC_DP_FAMILY_MUSTANG:
+		/* TODO: determine max firmware size for Mustang */
+		break;
+	case FU_KINETIC_DP_FAMILY_PUMA:
+		/* TODO: determine max firmware size for Puma */
+	default:
+		break;
```

Comment:
can we not populate these immediately?  I worry they'll get forgotten.

Reply:
I don't think these make much sense with the hierarchal firmware we have now. Removed.